$OpenBSD: patch-objects_drawin_c,v 1.3 2014/04/15 10:32:36 dcoppa Exp $

Fix a bug in luaA_drawin_set_shape_bounding() which forgot to include
the drawin's border in its size calculation
(part of upstream git commit 56c57979056ec23cbd4982f3b81f6d80895fbe46)

--- objects/drawin.c.orig	Sat Oct 12 18:48:49 2013
+++ objects/drawin.c	Tue Jan 14 10:36:22 2014
@@ -547,7 +547,9 @@ luaA_drawin_set_shape_bounding(lua_State *L, drawin_t 
     cairo_surface_t *surf = NULL;
     if(!lua_isnil(L, -1))
         surf = (cairo_surface_t *)lua_touserdata(L, -1);
-    xwindow_set_shape(drawin->window, drawin->geometry.width, drawin->geometry.height,
+    xwindow_set_shape(drawin->window,
+            drawin->geometry.width + 2*drawin->border_width,
+            drawin->geometry.height + 2*drawin->border_width,
             XCB_SHAPE_SK_BOUNDING, surf, -drawin->border_width);
     return 0;
 }
